Library/Homebrew/cmd/update.sh

```diff
@@ -599,7 +599,13 @@ EOS
         -d "$HOMEBREW_LIBRARY/LinkedKegs" ||
         (-n "$HOMEBREW_DEVELOPER" && -z "$HOMEBREW_UPDATE_PREINSTALL") ]]
   then
-    unset HOMEBREW_RUBY_PATH
+    if [[ -n $HOMEBREW_FORCE_VENDOR_RUBY &&
+          -x $HOMEBREW_LIBRARY/Homebrew/vendor/portable-ruby/current/bin/ruby ]]
+    then
+      export HOMEBREW_RUBY_PATH="$HOMEBREW_LIBRARY/Homebrew/vendor/portable-ruby/current/bin/ruby"
```

Comment:
This is already done by `utils/ruby.sh` when `brew update-report` is called.

Reply:
> This should never be the case. `utils/ruby.sh` should be finding the relevant Ruby from the PATH

`echo $PATH` right next to `HOMEBREW_RUBY_PATH="$(type -P ruby)"` in `utils/ruby.sh` -- it shows that PATH is `/usr/bin:/bin:/usr/sbin:/sbin` so it finds system Ruby only.

Reply:
`ruby.sh` is being run again so I don't see a reason to duplicate the logic here. It could do something with `HOMEBREW_PATH` if needed. @sjackman also had thoughts about this.

Reply:
> `ruby.sh` is being run again

Exactly. And in case of `update` command, `HOMEBREW_RUBY_PATH` is unset in two places: here (unconditionally for everyone) and then for non-developers in ruby.sh. Because of the unset here, `brew update` is the only command that fails on old systems where Ruby has to be bootstrapped and Homebrew pointed to it with `HOMEBREW_RUBY_PATH`...
Anyway, I'll move logic to ruby.sh